File: Modul-4/Soal-3/PRAK403-2410817220012-NOOR-KHALISA.c
#include <stdio.h>

void cetakBersilangan(int a, int b) {
    int i, j;

    if (a < b) {
        i = a;
        j = b;
        while (i <= b && j >= a) {
            printf("%d %d", i, j);
            if (i + 1 <= b || j - 1 >= a) {
                printf(" - ");
            }
            i++;
            j--;
        }
        printf("\n");
    } else {
        i = a;
        j = b;
        while (i >= b && j <= a) {
            printf("%d %d", i, j);
            if (i - 1 >= b || j + 1 <= a) {
                printf(" - ");
            }
            i--;
            j++;
        }
        printf("\n");
    }
}

int main() {
    int a, b;

    printf("Input pertama: ");
    scanf("%d", &a);
    printf("Input kedua: ");
    scanf("%d", &b);

    cetakBersilangan(a, b);

    return 0;
}

File: Modul-4/Soal-3/PRAK403-2410817220012-NOOR-KHALISA.py
def cetak_bersilangan(a, b):
    if a < b:
        i, j = a, b
        hasil = []
        while i <= b and j >= a:
            hasil.append(f"{i} {j}")
            i += 1
            j -= 1
    else:
        i, j = a, b
        hasil = []
        while i >= b and j <= a:
            hasil.append(f"{i} {j}")
            i -= 1
            j += 1
    print(" - ".join(hasil))

a = int(input("Input pertama: "))
b = int(input("Input kedua: "))

cetak_bersilangan(a, b)